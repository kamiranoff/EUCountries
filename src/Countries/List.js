import React from 'react';
import PropTypes from 'prop-types';
import { FlatList } from 'react-native';
import CountryItem from './CountryItem';

const List = ({ navigation: { state: { params } } }) => (
  <FlatList
    data={params.countries}
    renderItem={({ item }) => <CountryItem item={item} />}
    keyExtractor={(item, index) => `country-list${index}`}
  />
);

List.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
};

export default List;
