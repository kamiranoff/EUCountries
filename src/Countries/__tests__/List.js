import React from 'react';
import renderer from 'react-test-renderer';
import List from '../List';

const countries = [{ name: 'France', subregion: 'Western eu', capital: 'Paris', languages: [{ name: 'French' }] }]
const navigation = { navigate: jest.fn(), state: { params: { countries } } };

describe('List', () => {
  it('should render the component', () => {
    const wrapper = renderer.create(<List navigation={navigation} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
})