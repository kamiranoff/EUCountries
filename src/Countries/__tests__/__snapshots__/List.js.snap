// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render the component 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "capital": "Paris",
        "languages": Array [
          Object {
            "name": "French",
          },
        ],
        "name": "France",
        "subregion": "Western eu",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  invertStickyHeaders={undefined}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#eee",
            "margin": 10,
            "padding": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontWeight": "600",
            }
          }
        >
          France
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Western eu
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Paris
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Lang: 
          French
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
